tests/libgit/test_libgit_spec.lua

```diff
@@ -11,42 +31,160 @@ describe("Libgit2 core methods Test", function()
     lib.luagit2_SET_MWINDOW_MAPPED_LIMIT(4321)
-    new_val_MWINDOW_MAPPED_LIMIT = lib.luagit2_GET_MWINDOW_MAPPED_LIMIT()
+    local new_val_MWINDOW_MAPPED_LIMIT = lib.luagit2_GET_MWINDOW_MAPPED_LIMIT()
     assert.is.equal(4321,new_val_MWINDOW_MAPPED_LIMIT)
+
+  end)
+
+  it("Tests Setting and Getting Various Search Paths", function()
+    -- Tests luagit2_GET_SEARCH_PATH
+    -- for different options
+    assert.are.equal(Search_Paths.System_lvl,lib.luagit2_GET_SEARCH_PATH(1))
+    assert.are.equal(Search_Paths.Global_lvl,lib.luagit2_GET_SEARCH_PATH(2))
+    assert.are.equal(Search_Paths.XDG_lvl,lib.luagit2_GET_SEARCH_PATH(3))
+    assert.are.equal(Search_Paths.Program_data_lvl,lib.luagit2_GET_SEARCH_PATH(4))
+
+    -- reset the search paths
+    lib.luagit2_SET_SEARCH_PATH(1,Alternate_Path)
+    lib.luagit2_SET_SEARCH_PATH(2,Alternate_Path)
+    lib.luagit2_SET_SEARCH_PATH(3,Alternate_Path)
+    lib.luagit2_SET_SEARCH_PATH(4,Alternate_Path)
+
+    -- recheck for new paths
+    -- Tests luagit2_SET_SEARCH_PATH()
+    assert.are.equal(Alternate_Path,lib.luagit2_GET_SEARCH_PATH(1))
+    assert.are.equal(Alternate_Path,lib.luagit2_GET_SEARCH_PATH(2))
+    assert.are.equal(Alternate_Path,lib.luagit2_GET_SEARCH_PATH(3))
+    assert.are.equal(Alternate_Path,lib.luagit2_GET_SEARCH_PATH(4))
+
+  end)
+
+  it("Tests Setting and Getting Max Cache Size", function()
+
+    -- Check if Max Cache Size value is set correctly.
+    Current_Cache_Memory = lib.luagit2_GET_CACHED_MEMORY(1)
+    Allowed_Cache_memory = lib.luagit2_GET_CACHED_MEMORY(2)
+    assert.is.equal(Cache_Max_Size, Allowed_Cache_memory)
+
+    -- Since We dont know current Cache memory size,
+    -- We simply assert that it is an integer value
+    -- and is LESS THAN or EQUAL to Max Allowed Value
+    assert.is_number(Current_Cache_Memory)
+    assert(Current_Cache_Memory <= Allowed_Cache_memory)
+
+    -- Reset the Max Cache memory Size
+    -- And check again
+    local Max_Size = 4096
+    lib.luagit2_SET_CACHE_MAX_SIZE(Max_Size)
+    assert.is.equal(Max_Size,lib.luagit2_GET_CACHED_MEMORY(2))
+
+  end)
+
+  it("Tests Setting and Getting Template Path", function()
+    -- Tests luagit2_GET_TEMPLATE_PATH()
+    assert.is.equal(Template_path , lib.luagit2_GET_TEMPLATE_PATH())
+
+    -- Reset template path value and recheck
+    lib.luagit2_SET_TEMPLATE_PATH(Alternate_Path)
+    assert.is.equal(Alternate_Path,lib.luagit2_GET_TEMPLATE_PATH())
+  end)
+
+  it("Checks if Windows Share Mode is set", function()
+    -- Currently the test is run on Non windows system
+    -- So this share mode value may vary
+    -- Lets Simply Assert that it is a number
+    assert.is_number(lib.luagit2_GET_WINDOWS_SHAREMODE())
+  end)
+
+  it("Checks Enable and Disable for various options", function()
+    -- We simply call the enable functions.
+    -- If there will be any error, The program
+    -- will shutdown itself.
+
+     -- Enable Various possible options
+    lib.luagit2_ENABLE_CACHING(Enable)
```

Comment:
See this comment in a different PR 
https://github.com/SatyendraBanjare/luagit2/pull/6#pullrequestreview-134080487

Reply:
Ok :smile: